packages/editor-ui/src/components/RunData.vue

```diff
@@ -759,7 +758,10 @@ export default defineComponent({
 		this.hidePinDataDiscoveryTooltip();
 	},
 	computed: {
-		...mapStores(useNodeTypesStore, useNDVStore, useWorkflowsStore),
+		...mapStores(useNodeTypesStore, useNDVStore, useWorkflowsStore, useSourceControlStore),
+		isReadOnlyRoute() {
+			return this.$route?.meta?.readonly === true;
+		},
```

Comment:
Yes, although we would still need to pass the router, or restrict this composable to be used only if there's component instance(to prevent nested usage in composable and `useRoute` being undefined)